lib/usd/translators/shading/mtlxBaseWriter.cpp

```diff
@@ -279,43 +295,55 @@ MtlxUsd_BaseWriter::AddSwizzleConversion(const SdfValueTypeName& destType, UsdAt
     // There are 6 types to handle. Not enough that a sorted container would help. Grouping them
     // will allow finding any item in maximum 10 iterations by skipping whole chunks. Putting Float3
     // and Float4 last since they are the least likely to be searched for.
+    //
+    // An empty swizzle signals the use a "convert" node.
+    //
+    // TODO: MaterialX 1.39 will no longer support swizzles, requiring use of two nodes (separate +
+    //       combine) to do the work of one.
+    // NOTE: in 1.38 the separate node is 1) implemented using a swizzle 2) not inlined, so we will
+    //       wait until 1.39 to implement that change.
+    // This requires making sure that both USD and MaterialX work correctly with multi-output nodes.
+    //   Requires at least the upcoming USD 22.05 and MaterialX 1.38.4:
+    //     https://github.com/PixarAnimationStudios/USD/commit/9bcc8baa
+    //     https://github.com/AcademySoftwareFoundation/MaterialX/pull/853
+    //
     typedef std::vector<_SwizzleData> _SwizzleMap;
     static const _SwizzleMap          _swizzleMap = _SwizzleMap {
-        { SdfValueTypeNames->Float, SdfValueTypeNames->Float2, "float", "vector2", "xx" },
```

Comment:
Trying to lower the number of swizzle nodes by using `convert` nodes that expose the exact same semantics.